### PR TITLE
Upgrading TF ROCm CI to Ubuntu 18.04

### PR DESCRIPTION
**Pull request by parallelo**
Upgrading TF ROCm CI to Ubuntu 18.04.  Also needed to upgrade clang pkgs and removed an old CMake PPA.

### CODE REVIEW

**Comment by ROCm-Apps-Test** [comment]
Can one of the admins verify this patch?

File: tensorflow/tools/ci_build/Dockerfile.rocm [hunk 1]
```diff
@@ -102,8 +102,8 @@ RUN touch ${ROCM_PATH}/.info/version
 COPY install/*.sh /install/
 ARG DEBIAN_FRONTEND=noninteractive
 RUN /install/install_bootstrap_deb_packages.sh
-RUN add-apt-repository -y ppa:openjdk-r/ppa && \
-    add-apt-repository -y ppa:george-edison55/cmake-3.x
+RUN add-apt-repository -y ppa:openjdk-r/ppa
```
Comment (deven-amd) [inline tensorflow/tools/ci_build/Dockerfile.rocm]: can we get rid of this line too?

LGTM otherwise

Comment (parallelo) [reply tensorflow/tools/ci_build/Dockerfile.rocm]: Yep, you are right.  That line probably isn't needed anymore.  OpenJDK is installed in the `install_deb_packages.sh` script.

**Comment by parallelo** [comment]
Four errors in the Ubuntu 18.04 `rocm` build:
```
//tensorflow/python/kernel_tests:cwise_ops_test                         [0m[31m[1mTIMEOUT[0m in 3 out of 3 in 915.0s
//tensorflow/python/keras/distribute:keras_embedding_model_correctness_test [0m[31m[1mFAILED[0m in 3 out of 3 in 216.8s
//tensorflow/python/kernel_tests/linalg/sparse:csr_sparse_matrix_sparse_mat_mul_grad_test [0m[31m[1mFAILED[0m in 3 out of 3 in 120.7s
//tensorflow/python/kernel_tests/linalg/sparse:csr_sparse_matrix_sparse_mat_mul_grad_test_gpu [0m[31m[1mFAILED[0m in 3 out of 3 in 90.5s
```

**Comment by parallelo** [comment]
Last night's _Ubuntu 16.04_ internal builds showed two errors:
```
//tensorflow/python/keras/distribute:keras_embedding_model_correctness_test [0m[31m[1mFAILED[0m in 3 out of 3 in 915.0s
//tensorflow/python/kernel_tests/linalg/sparse:csr_sparse_matrix_sparse_mat_mul_grad_test [0m[31m[1mFAILED[0m in 3 out of 3 in 113.3s
```

Let's check for any variability in these failures.  

Jenkins: retest this please

**Comment by parallelo** [comment]
The `rocm` unit tests are now clear:  
```
Executed 1446 out of 1446 tests: 1446 tests pass.
```

Merging this PR.

**Pull request merged**